### DISCUSSION TITLE
Open One or Two Widgets From Widget Icon

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9629] [Q&A]
Hello - I wanted to know if there is an example where I could have the ability to click on a widget icon (that is defaulted to the left) and open one or multiple widgets in other "panes" (main, right, etc.) at the same time?

**Comment by vince-fugnitto** [comment]
@sboudreaux96 there is no example that I am aware of.

You can use the `openView` functionality of your widgets to possibly register a command which will open the appropriate widgets in a layout you require with the use of the available options:
- [open-view](https://eclipse-theia.github.io/theia/docs/next/classes/core.abstractviewcontribution-2.html#openview)
- [open-view-arguments](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.openviewarguments-2.html)
- [widget-options](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.applicationshell-2.widgetoptions.html)

**Reply by sboudreaux96** [reply]
I am currently trying to implement this but am having issues on how to use open-view from one widget to open another widget. Would it be possible for you to provide some general steps that would need to be done?

**Reply by vince-fugnitto** [reply]
In general, you can:
- create a `CommandContribution`
- inject the necessary `AbstractViewContribution`s you have of your views (giving you access to their `openView` implementation)
- create a command that calls `openView` on your injection contributions with the appropriate options you require for your use-case

You can play around with the idea, the goal is to access `openView` which is accessible through the view's respective `AbstractViewContribution`, and call it in a command, or another view's `openView`.
